app/upgrades/vNEXT.go

```diff
@@ -31,13 +74,112 @@ var Upgrade_NEXT = Upgrade{
+			// steps (INIT → TRY → ACK → CONFIRM), each of which occurs in separate blocks on
+			// each chain. As such, this timeout should account for at least 4 relayed IBC messages,
+			// which means it must exceed:
+			//
+			//     timeout ≥ 4 × slower_chain_block_time × adjustment_factor
+			//
+			// For localnets, a value between 60–300 seconds (e.g. 300_000_000_000 ns) is typically
+			// sufficient to accommodate relayer delays. A zero value is considered invalid unless
+			// a non-zero height-based timeout is also set.
+			ibcChannelParamUpgradeTimeoutTimestamp, err := time.ParseDuration(
+				fmt.Sprintf(durationSecondsFormat, upgradeTimeoutSeconds),
+			)
+			if err != nil {
+				return err
+			}
+
+			ibcChannelParams := channeltypes.Params{
+				UpgradeTimeout: channeltypes.Timeout{
+					Height: ibcclienttypes.Height{
+						RevisionNumber: IbcChannelParamUpgradeTimeoutRevisionNumber,
+						RevisionHeight: IbcChannelParamUpgradeTimeoutRevisionHeight,
+					},
+					Timestamp: uint64(ibcChannelParamUpgradeTimeoutTimestamp.Nanoseconds()),
+				},
+			}
+			fmt.Printf("IBC channel params: %+v\n", ibcChannelParams)
+
+			// IBC client params
+			// See: https://ibc.cosmos.network/params/params.md/
+			ibcClientParams := ibcclienttypes.Params{
+				AllowedClients: IbcClientParamAllowedClients,
+			}
+			fmt.Printf("IBC client params: %+v\n", ibcClientParams)
+
+			// IBC transfer params
+			// See: https://ibc.cosmos.network/v8/apps/transfer/params/
+			ibcTransferParams := ibctransfertypes.Params{
+				SendEnabled:    IbcTransferParamSendEnabled,
+				ReceiveEnabled: IbcTransferParamReceiveEnabled,
+			}
+			fmt.Printf("IBC transfer params: %+v\n", ibcTransferParams)
+
+			// IBC interchain accounts host params
+			ibcIcaHostParams := icahosttypes.Params{
+				HostEnabled:   IbcIcaHostParamHostEnabled,
+				AllowMessages: IbcIcaHostParamAllowMessages,
+			}
+			fmt.Printf("IBC interchain accounts host params: %+v\n", ibcIcaHostParams)
+
+			// IBC interchain accounts controller params
+			// See: https://ibc.cosmos.network/v8/apps/interchain-accounts/parameters/
+			ibcIcaControllerParams := icacontrollertypes.Params{
+				ControllerEnabled: IbcIcaControllerParamControllerEnabled,
+			}
+			fmt.Printf("IBC interchain accounts controller params: %+v\n", ibcIcaControllerParams)
+
+			// Set IBC core params
+			// - connection
+			// - channel
+			// - client
+			keepers.IBCKeeper.ConnectionKeeper.SetParams(sdkCtx, ibcConnectionParams)
+			keepers.IBCKeeper.ChannelKeeper.SetParams(sdkCtx, ibcChannelParams)
+			keepers.IBCKeeper.ClientKeeper.SetParams(sdkCtx, ibcClientParams)
+
+			// Set IBC transfer params
+			keepers.TransferKeeper.SetParams(sdkCtx, ibcTransferParams)
+
+			// Set IBC interchain accounts host & controller params
+			keepers.ICAHostKeeper.SetParams(sdkCtx, ibcIcaHostParams)
+			keepers.ICAControllerKeeper.SetParams(sdkCtx, ibcIcaControllerParams)
```

Comment:
Have you tested this upgarde on LocalNet?

I don't really know how to review this, and it is a very critical piece of code.